FantasyUniverse/Scenes/Quiz/QuizView.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  QuestionView.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 25.12.2021.
+//
+
+import SwiftUI
+
+struct QuizView: View {
+    
+    @Binding var set: QuestionSet
+    @StateObject var viewModel = QuizViewModel()
+    @Environment(\.presentationMode) var present
+    
+    var body: some View {
+        
+        AnimatedBackground(animationName: "day-background") {
+            if viewModel.questionsData.isEmpty {
+                LoadingView()
+            } else {
+                if viewModel.answered == viewModel.questionsData.count {
+                    // Finish screen
+                    scoreView
+                } else {
+                    // Game screen
+                    VStack(alignment: .center) {
+                        topProgress
+                        questionView
+                    }
+                    .padding(.horizontal, 5)
+                }
+            }
+            
+        }
+        .navigationBarHidden(true)
+        .onAppear {
+            viewModel.getQuestions(set: set)
+        }
+        .onDisappear {
+            cleanUp()
```

Comment:
You shouldn't need to clean it up, the instance should be deallocated for you on disappear